pkgs/development/compilers/swift/default.nix

```diff
@@ -0,0 +1,261 @@
+{ stdenv
+, cmake
+, coreutils
+, glibc
+, which
+, perl
+, libedit
+, ninja
+, pkgconfig
+, sqlite
+, swig
+, bash
+, libxml2
+, llvm
+, clang
+, python
+, ncurses
+, libuuid
+, libbsd
+, icu
+, autoconf
+, libtool
+, automake
+, libblocksruntime
+, curl
+, rsync
+, git
+, libgit2
+, binutils
+, fetchFromGitHub
+, paxctl
+, findutils
+#, systemtap
+}:
+
+let
+  v_major = "3.1";
+  version = "${v_major}-RELEASE";
+  version_friendly = "${v_major}";
+
+  tag = "refs/tags/swift-${version}";
+  fetch = { repo, sha256, fetchSubmodules ? false }:
+    fetchFromGitHub {
+      owner = "apple";
+      inherit repo sha256 fetchSubmodules;
+      rev = tag;
+      name = "${repo}-${version}-src";
+    };
+
+sources = {
+    clang = fetch {
+      repo = "swift-clang";
+      sha256 = "0820mx1ghfnk4p5595r1f313y1699jwi61zghfbwak5wgwgy7n4x";
+    };
+    llvm = fetch {
+      repo = "swift-llvm";
```

Comment:
I was unable to convince their build scripts to use an external LLVM.  I believe their versions of these components are very similar but haven't tried sorting out differences (especially since I don't think they're based on a particular fixed release).  Without more insight into the differences I'd be reluctant to use anything other than their vendor'd LLVM due to concerns about possible subtle behavioral differences from other "Swift 3.1" implementations.

That said, even breaking this huge build into pieces would be awfully nice :).